### PR TITLE
fix(anvil): avoid mutable/immutable borrow conflict

### PR DESCRIPTION
**Pull request by axic**
```
warning: cannot borrow `*self` as mutable because it is also borrowed as immutable
  --> anvil/src/eth/backend/mem/in_memory_db.rs:65:13
   |
63 |               let old_account = self.inner.accounts.get(&addr);
   |                                 ------------------------------ immutable borrow occurs here
64 | 
65 | /             self.insert_account(
66 | |                 addr,
67 | |                 AccountInfo {
68 | |                     balance: account.balance,
...  |
78 | |                         old_account.map(|a| a.info.nonce).unwrap_or_default(),
   | |                         ----------- immutable borrow later used here
...  |
81 | |                 },
82 | |             );
   | |_____________^ mutable borrow occurs here
   |
   = note: `#[warn(mutable_borrow_reservation_conflict)]` on by default
   = warning: this borrowing pattern was not meant to be accepted, and may become a hard error in the future
   = note: for more information, see issue #59159 <https://github.com/rust-lang/rust/issues/59159>
 ```

This just fixes this warning, which may become an error in later Rust versions.

### CODE REVIEW

File: anvil/src/eth/backend/mem/in_memory_db.rs [hunk 1]
```diff
@@ -61,6 +61,7 @@ impl Db for MemDb {
     fn load_state(&mut self, state: SerializableState) -> DatabaseResult<bool> {
         for (addr, account) in state.accounts.into_iter() {
             let old_account = self.inner.accounts.get(&addr);
+            let old_account_nonce = old_account.map(|a| a.info.nonce).unwrap_or_default();
```
Comment (axic) [inline anvil/src/eth/backend/mem/in_memory_db.rs:64]: Can merge these into a single expression.

**Review by gakonst — APPROVED** [review]

**Review by mattsse — CHANGES_REQUESTED** [review]
TIL

let's just move the entire thing out of the call so that we have

`let nonce = ...`

**Comment by axic** [comment]
Pushed the changes if I understood correctly. Not sure however if this last commit makes sense, I think it is clearer without it.

**Review by mattsse — APPROVED** [review]
ty!

**Review by gakonst — APPROVED** [review]

**Comment by axic** [comment]
Thanks for the quick review!

**Pull request merged**